### PR TITLE
Implement garbage collection

### PR DESCRIPTION
**Pull request by stefanprodan**
Changes:
- add `GarbageCollect` predicate to `KustomizationReconciler`
- remove the Kubernetes objects previously applied on the cluster, based on `spec.prune` label select
- add garbage collection spec to docs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,9 +44,26 @@ type KustomizationSpec struct {
...
 
+### Garbage collection
+
+Garbage collection means that the Kubernetes objects that were previously applied on the cluster
+but are missing from the current apply, will be removed. Garbage collection is also performed when a Kustomization
+object is deleted, triggering a removal of all Kubernetes objects previously applied on the cluster.
+
+When garbage collection is enabled, all Kubernetes objects must have a common label that matches the `prune` selector.
+
+For example, `prune: env=dev` requires a kustomization.yaml with:
```
Comment (hiddeco) [inline README.md:59]:
```suggestion
For example, `prune: env=dev` requires a `kustomization.yaml` with:
```

File: api/v1alpha1/kustomization_types.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ type KustomizationSpec struct {
 	// +required
 	Path string `json:"path"`
 
-	// Label selector used for prune operations, e.g. env=staging.
+	// Label selector used for garbage collection.
 	// +kubebuilder:validation:Pattern="^.*=.*$"
 	// +optional
 	Prune string `json:"prune,omitempty"`
```
Comment (hiddeco) [inline api/v1alpha1/kustomization_types.go:34]: Is a single label sufficient, or would a `[]string` be better?

Comment (stefanprodan) [reply api/v1alpha1/kustomization_types.go:34]: label selectors are not an array but an an expression e.g.:
`prune: "environment in (production),tier in (frontend)"`

Comment (stefanprodan) [reply api/v1alpha1/kustomization_types.go:34]: I've added the Kubernetes docs link to the readme.

**Pull request merged**